lib/web_ui/lib/src/engine/semantics/tappable.dart

```diff
@@ -33,42 +30,44 @@ class Button extends PrimaryRoleManager {
 /// the browser may not send us pointer events. In that mode we forward HTML
 /// click as [ui.SemanticsAction.tap].
 class Tappable extends RoleManager {
-  Tappable(SemanticsObject semanticsObject)
-      : super(Role.tappable, semanticsObject);
+  Tappable(SemanticsObject semanticsObject) : super(Role.tappable, semanticsObject) {
+    _clickListener = createDomEventListener((DomEvent click) {
+      final bool shouldDeduplicateClickEvent = PointerBinding.instance!.clickDebouncer.onClick(click);
+
+      if (!_isListening || shouldDeduplicateClickEvent) {
+        return;
+      }
```

Comment:
When `!_isListening`, do we need to call `clickDebouncer.onClick(click)`?

i.e. something like this:
```suggestion
      if (!_isListening) {
        return;
      }
      final bool shouldDeduplicateClickEvent = PointerBinding.instance!.clickDebouncer.onClick(click);

      if (shouldDeduplicateClickEvent) {
        return;
      }
```

Reply:
In either case (`isListeening` being true or false) the node should tell the debouncer to stop debouncing because the click gesture has ended. That's why `onClick` is called unconditionally. Otherwise, the debouncer will have to wait for the timer to expire before going back to normal state.

However, barring the [issue with scrollables](https://github.com/flutter/flutter/issues/130162) it should be a near-impossible situation that `isListening` is false _and_ the debouncer is debouncing events. This is because the debouncer only start debouncing if `target.hasAttribute('flt-tappable')` is true. If the node is not listening, its DOM element won't have the `flt-tappable` attribute (see changes in `Tappable.update` method).

## But!

Maybe there should be a second callback for the "not listening" case. Instead of `onClick`, maybe we should have `onNotListening`. The difference is that the debouncer would flush the events to the framework instead of dropping them.

🤔